client/src/components/Collections/SampleSheetWizard.vue

```diff
@@ -0,0 +1,507 @@
+        if (pastedData.length === 0) {
+            return false;
+        }
+        for (const row of pastedData) {
+            if (row.length !== 2) {
+                return false; // we expect exactly two columns for each paired data.
+            }
+        }
+        return true;
+    } else {
+        return false;
+    }
+});
+
+function guessUriFilename(uri: string): string {
+    const parts = uri.split("/");
+    const last_part = parts[parts.length - 1] ?? uri;
+    if (last_part.indexOf("?") !== -1) {
+        // remove the query string
+        return last_part.split("?")[0] as string;
+    } else {
+        return last_part;
+    }
+}
+
+function adaptUriToAutoPair(uri: string): { name: string; uri: string } {
+    return { name: guessUriFilename(uri), uri };
+}
+
+const toAutoPair = computed(() => {
+    if (props.collectionType === "sample_sheet:paired" || props.collectionType === "sample_sheet:paired_or_unpaired") {
+        if (pastedDataLooksPrePaired.value) {
+            // don't auto-pair anything - the data was pasted in two clean columns - take it as is.
+            return undefined;
+        }
+        if (sourceFrom.value == "pasted_table") {
+            const uris: string[] = [];
+            for (const row of pasteData.value) {
+                uris.push(...row);
+            }
+            return uris.map(adaptUriToAutoPair);
+        } else {
+            return undefined;
+        }
+    } else {
+        return undefined;
+    }
+});
+
+const allPaired = computed<boolean>(() => {
+    return !!(toAutoPair.value && countPaired.value === toAutoPair.value.length / 2);
+});
+
+watch(toAutoPair, (value) => {
+    if (value !== undefined) {
+        autoPair(value);
+    }
+});
+
+const collectionCreator = ref<CollectionCreatorComponent>();
+
+function submit() {
+    if (collectionCreator.value) {
+        attemptCreate(collectionCreator);
+    }
+}
+
+const isSimpleSampleSheet = computed(() => {
+    // we can do more with a simple list of URLs that don't need to be paired..
+    return props.collectionType === "sample_sheet";
+});
+
+const fetchJobId = ref<string | undefined>(undefined);
+const waitingOnCollectionCreateApi = ref<boolean>(false);
+const collectionCreateError = ref<string | undefined>(undefined);
+const collectionCreated = ref<boolean>(false);
+const { getJob, pollJobUntilTerminal } = useJobStore();
+
+const job = computed(() => {
+    if (fetchJobId.value) {
+        const jobId = fetchJobId.value;
+        const job = getJob(jobId);
+        if (job) {
+            return job;
+        }
+    }
+    return undefined;
+});
+
+async function onFetchTarget(fetchTarget: HdcaUploadTarget) {
+    const fetchPayload: FetchDataPayload = {
+        history_id: currentHistoryId.value as string,
+        targets: [fetchTarget],
+    };
+    try {
+        const jobId = await fetch(fetchPayload);
+        fetchJobId.value = jobId;
+        pollJobUntilTerminal({ id: jobId });
```

Comment:
Why do we need to do this ? Once we have the job we also have the outputs and we can go ahead ?

Reply:
If it's only to track progress that's fine, I thought we might be waiting unti the job is complete.